airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/abstract_oauth.py

```diff
@@ -96,14 +96,45 @@ def _get_refresh_access_token_response(self):
         except Exception as e:
             raise Exception(f"Error while refreshing access token: {e}") from e
 
-    def refresh_access_token(self) -> Tuple[str, int]:
+    def refresh_access_token(self) -> Tuple[str, Union[str, int]]:
         """
-        Returns the refresh token and its lifespan in seconds
+        Returns the refresh token and its lifespan
 
-        :return: a tuple of (access_token, token_lifespan_in_seconds)
+        :return: a tuple of (access_token, token_lifespan)
         """
         response_json = self._get_refresh_access_token_response()
-        return response_json[self.get_access_token_name()], int(response_json[self.get_expires_in_name()])
+
+        return response_json[self.get_access_token_name()], response_json[self.get_expires_in_name()]
+
+    def _parse_token_lifespan(self, value: Union[str, int]) -> pendulum.DateTime:
+        """
+        Return the lifespan of refresh token
+
+        :return: lifespan datetime
+        """
+
+        if self.token_expiry_is_time_of_expiration:
+            if not self.token_expiry_date_format:
+                raise ValueError(f"Invalid token expiry date format {self.token_expiry_date_format}; a string is required.")
+            return pendulum.from_format(value, self.token_expiry_date_format)
+        else:
+            return pendulum.now().add(seconds=int(float(value)))
+
+    @property
+    def token_expiry_is_time_of_expiration(self) -> bool:
+        """
+        Indicates that the Token Expiry returns the date until which the token will be valid, not the amount of time it will be valid.
+        """
+
+        return False
+
+    @property
+    def token_expiry_date_format(self) -> Optional[str]:
+        """
+        Format of the datetime; provide it if expires_in is returned in datetime instead of seconds
```

Comment:
```suggestion
        Format of the datetime; exists it if expires_in is returned as the expiration datetime instead of seconds until it expires
```